includes/class-wcpdf-admin.php

```diff
@@ -798,26 +816,56 @@ public function regenerate_document() {
 
 		try {
 			$document = wcpdf_get_document( $document_type, wc_get_order( $order_id ) );
-			if ( !empty($document) && $document->exists() ) {
-				// perform legacy date fields replacements check
-				if( isset( $form_data["_wcpdf_{$document->slug}_date"] ) && ! is_array( $form_data["_wcpdf_{$document->slug}_date"] ) ) {
-					$form_data = $this->legacy_date_fields_replacements( $form_data, $document->slug );
+
+			if( ! empty( $document ) ) {
+
+				// on regenerate
+				if( $action_type == 'regenerate' && $document->exists() ) {
+					// perform legacy date fields replacements check
+					if( isset( $form_data["_wcpdf_{$document->slug}_date"] ) && ! is_array( $form_data["_wcpdf_{$document->slug}_date"] ) ) {
+						$form_data = $this->legacy_date_fields_replacements( $form_data, $document->slug );
+					}
+
+					// save document data
+					$document_data = $this->process_order_document_form_data( $form_data, $document->slug );
+					$document->regenerate( $order, $document_data );
+
+					$response      = array(
+						'message' => $document->get_type()." regenerated",
+					);
+
+				// on save
+				} elseif( $action_type == 'save' ) {
+					$form_data     = stripslashes_deep( $form_data );
+					$document_data = $this->process_order_document_form_data( $form_data, $document->slug );
```

Comment:
I think the data processing part should be the same for both action types, so instead of applying the above two lines, move these lines out of the if-statement: https://github.com/wpovernight/woocommerce-pdf-invoices-packing-slips/blob/ddb43fca5712729524557d2373098a3a085cc774/includes/class-wcpdf-admin.php#L824-L830